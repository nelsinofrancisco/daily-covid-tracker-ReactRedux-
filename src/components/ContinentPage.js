import ContinentBanner from './continent-components/ContinentBanner';
import ContinentList from './continent-components/ContinentList';

const ContinentPage = () => (
  <>
    <ContinentBanner />
    <ContinentList />
  </>
);

export default ContinentPage;
