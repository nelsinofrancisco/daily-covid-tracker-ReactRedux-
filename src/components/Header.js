import { useRouteMatch } from 'react-router';
import { HiOutlineChevronLeft } from 'react-icons/hi';
import { BsGearWide } from 'react-icons/bs';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';

const headerStyles = {
  container: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: 'var(--primary-bc-color)',
    width: '100%',
    padding: '0.5rem',
  },
  title: {
    fontFamily: 'var(--title-font)',
    fontWeight: '300',
  },
};

const titleHandler = (path, continent, country, region) => {
  // Implement route to this later >>
  if (path.includes('/region/')) {
    return `${region}/cases`;
  }

  if (path.includes('/country/')) {
    return `${country}/cases`;
  }

  if (path.includes('/continent/')) {
    return `${continent}/cases`;
  }

  return 'Home';
};

const backLinkHandler = (path, continent, country) => {
  // Implement route to this later >>
  if (path.includes('/region/')) {
    return `/continent/${continent}/country/${country}`;
  }

  if (path.includes('/country/')) {
    return `/continent/${continent}/`;
  }

  if (path.includes('/continent/')) {
    return '/';
  }

  return '/';
};

const Header = () => {
  const { path } = useRouteMatch();
  const { continent, country, region } = useSelector((state) => state.covidData);

  const title = titleHandler(path, continent, country, region);

  return (
    <div style={headerStyles.container}>
      {
        (path === '/') ? (<span style={{ width: '16px' }} />) : (
          <NavLink
            to={backLinkHandler(path, continent, country)}
            className="active"
          >
            {' '}
            <HiOutlineChevronLeft />
          </NavLink>
        )
      }
      <p style={headerStyles.title}>{title}</p>
      <BsGearWide />
    </div>
  );
};

export default Header;
