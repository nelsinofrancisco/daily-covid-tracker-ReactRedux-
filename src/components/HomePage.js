import HomeBanner from './home-components/HomeBanner';
import HomeStats from './home-components/HomeStats';

const HomePage = () => (
  <>
    <HomeBanner />
    <HomeStats />
  </>
);

export default HomePage;
