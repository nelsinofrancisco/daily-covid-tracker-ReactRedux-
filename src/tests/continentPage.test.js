import React from 'react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';
import {
  filterCountries, filterCountry, getCovidData, selectContinent,
} from '../redux/covidData/covidDataSlice';
import Header from '../components/Header';
import HomePage from '../components/HomePage';
import ContinentPage from '../components/ContinentPage';

jest.mock('./fetchMock.js');

describe('Home page tests', () => {
  test('HomePage matches snapshot', async () => {
    await store.dispatch(getCovidData());
    const homePage = act(async () => renderer
      .create(
        <Provider store={store}>
          <Router>
            <Header />
            <HomePage />
          </Router>
        </Provider>,
      )
      .toJSON());
    expect(homePage).toMatchSnapshot();
  });

  describe('Link Interaction', () => {
    test('Check how many continent links are in the page => should be 6', () => {
      act(() => {
        render(
          <Provider store={store}>
            <Router>
              <Header />
              <HomePage />
            </Router>
          </Provider>,
        );
      });
      screen.queryAllByRole('link').forEach((role) => expect(role).toBeInTheDocument());
      expect(screen.queryAllByRole('link').length).toBe(6);
    });
    test('Fire continent view more', () => {
      act(() => {
        render(
          <Provider store={store}>
            <Router>
              <Header />
              <HomePage />
            </Router>
          </Provider>,
        );
      });
      fireEvent.select(screen.getByText('South America'));
      expect(screen.getByText('South America')).toBeInTheDocument();
    });
    test('Test if new page is rendered', async () => {
      await store.dispatch(selectContinent('South America'));
      await store.dispatch(filterCountries('South America'));
      const homePage = act(async () => renderer
        .create(
          <Provider store={store}>
            <Router>
              <Header />
              <ContinentPage />
            </Router>
          </Provider>,
        )
        .toJSON());
      expect(homePage).toMatchSnapshot();
    });
  });
  describe('Test Filter', () => {
    test('Test match by Brazil => since there is only one brazil country this should have a length of 1', async () => {
      await store.dispatch(filterCountry('brazil'));
      act(async () => {
        await render(
          <Provider store={store}>
            <Router>
              <Header />
              <ContinentPage />
            </Router>
          </Provider>,
        );
      });
      expect(screen.queryAllByRole('link').length).toBe(1);
    });
  });
});
