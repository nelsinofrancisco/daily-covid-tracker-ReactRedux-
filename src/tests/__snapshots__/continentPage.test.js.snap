// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page tests HomePage matches snapshot 1`] = `
Object {
  "then": [Function],
}
`;

exports[`Home page tests Link Interaction Test if new page is rendered 1`] = `
Object {
  "then": [Function],
}
`;
